### DISCUSSION TITLE
How to use window.localStorage with prerender

### DISCUSSION THREAD

**Discussion by PrettyCoffee** [preactjs/wmr#954] [Q&A] [answered]
I am trying to build a theme toggle that saves the currently selected theme into the local storage.
When executing `wmr` or `wmr build`, everything works as expected.
But when trying to build it with `wmr build --prerender`, it will always throw an error:
```
The following error was thrown during prerendering:
ReferenceError: window is not defined
```
This is not what I would expect, but I don't know what I am doing wrong or if this is something to be expected?

Reproduction:
1. `npm init wmr local-storage-test`
2. Add this in `public/index.js`: 
    ```
    window.localStorage.setItem("key", "value")
    ```
3. run `npm run build`

The code can also be added within the Home component to trigger the error.

**Comment by rschristian** [comment, marked as answer]
`window` is not accessible in Node, where prerendering happens. You'll need to use a window check/guard:

```js
if (typeof window !== 'undefined') {
    window.localStorage.setItem("key", value);
}
```

**Reply by PrettyCoffee** [reply]
Aaah, that makes a lot of sense. Thank you very much for this blazing fast answer! :)

**Reply by rschristian** [reply]
No problem, and if it helps make things clearer, here's a [`useLocalStorage` hook that I had in one of my projects](https://github.com/rschristian/diff-compare/blob/28daeb6f6185962fef3c08b8f93530928182f294/public/hooks/useLocalStorage.ts). It's really just a matter of inserting that check before using browser globals.
